### PR TITLE
azurerm_nginx_deployment - remove default `capacity` for v4.0

### PR DESCRIPTION
**Pull request by ryepup**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

The default value has caused headaches around autoscaling and new SKUs, requiring `ignore_changes` workarounds and making it harder for users to express their desired state.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

Manually tested locally a variety of cases:

| `sku` | `capacity` | `auto_scale_profiles` | outcome |
|-|-|-|-|
| `basic_Monthly` | absent | absent | PUT request does not include any scaling |
| `basic_Monthly` | present | absent | error during apply, but before PUT | 
| `basic_Monthly` | absent | present | error during apply, but before PUT | 
| `basic_Monthly` | present | present | error during planning | 
| `standard_Monthly` | absent | absent | error during apply, but before PUT |
| `standard_Monthly` | present | absent | PUT request includes `capacity` | 
| `standard_Monthly` | absent | present | PUT request includes `autoScaleProfiles` | 
| `standard_Monthly` | present | present | error during planning | 

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_nginx_deployment` -  remove default capacity for v4.0 [GH-27027]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [x] Breaking Change

## Related Issue(s)

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/nginx/nginx_deployment_resource.go [hunk 1]
```diff
@@ -439,13 +440,29 @@ func (m DeploymentResource) Create() sdk.ResourceFunc {
...
-					Capacity: pointer.FromInt64(model.Capacity),
+				if model.Capacity > 0 && !isBasicSKU {
+					prop.ScalingProperties = &nginxdeployment.NginxDeploymentScalingProperties{
+						Capacity: pointer.FromInt64(model.Capacity),
+					}
+				}
+			} else {
+				hasScaling := (model.Capacity > 0 || len(model.AutoScaleProfile) > 0)
+				if isBasicSKU && hasScaling {
+					return fmt.Errorf("basic SKUs are incompatible with `capacity` or `auto_scale_profiles`")
```
Comment (valyria257) [inline internal/services/nginx/nginx_deployment_resource.go:456]: Do we need to error here or do we want our API/future dry-run to handle this validation?

File: website/docs/r/nginx_deployment.html.markdown [hunk 2]
```diff
@@ -88,13 +88,13 @@ The following arguments are supported:
 
 * `sku` - (Required) Specifies the NGINX Deployment SKU. Possible values are `standard_Monthly` and `basic_Monthly`. Changing this forces a new resource to be created.
 
--> **NOTE:** If you are setting the `sku` to `basic_Monthly`, setting a non-default `capacity` or `auto_scale_profile` will fail. You should use [Terraform's `ignore_changes` functionality](https://www.terraform.io/language/meta-arguments/lifecycle#ignore_changes) to ignore changes to the `capacity` field.
+-> **NOTE:** If you are setting the `sku` to `basic_Monthly`, you cannot specify a `capacity` or `auto_scale_profile`, basic plans do not support scaling. Other `sku`s require either `capacity` or `auto_scale_profile`. If you're using `basic_Monthly` with deployments created before v4.0, you may need to use [Terraform's `ignore_changes` functionality](https://www.terraform.io/language/meta-arguments/lifecycle#ignore_changes) to ignore changes to the `capacity` field.
```
Comment (valyria257) [inline website/docs/r/nginx_deployment.html.markdown:91]: Super nit:
```suggestion
-> **NOTE:** If you are setting the `sku` to `basic_Monthly`, you cannot specify a `capacity` or `auto_scale_profile`; basic plans do not support scaling. Other `sku`s require either `capacity` or `auto_scale_profile`. If you're using `basic_Monthly` with deployments created before v4.0, you may need to use [Terraform's `ignore_changes` functionality](https://www.terraform.io/language/meta-arguments/lifecycle#ignore_changes) to ignore changes to the `capacity` field.
```

File: internal/services/nginx/nginx_deployment_resource.go (hunk 1 above)
Comment (ryepup) [reply internal/services/nginx/nginx_deployment_resource.go:456]: I think we want the error here for now, and I'm gonna add it to the update function as well. You're right that the API handles this, but until that "dry-run" feature gets implemented server side we can end up with some bad data. I figured this validation is unlikely to change, so OK to duplicate between client and server.

File: website/docs/r/nginx_deployment.html.markdown (hunk 2 above)
Comment (ryepup) [reply website/docs/r/nginx_deployment.html.markdown]: done

**Review by stephybun — APPROVED** [review]
Thanks @ryepup LGTM 👍

**Pull request merged**